zcash_primitives/src/transaction/sighash_v5.rs

```diff
@@ -170,22 +179,20 @@ pub fn v5_signature_hash<
     signable_input: &SignableInput<'_>,
     txid_parts: &TxDigests<Blake2bHash>,
 ) -> Blake2bHash {
+    // The caller must always provide the transparent digests if the transaction has a
+    // transparent component.
+    assert!(tx.transparent_bundle.is_some() ^ txid_parts.transparent_digests.is_none());
```

Comment:
```suggestion
    // The caller must provide the transparent digests if and only if the transaction has a
    // transparent component.
    assert_eq!(tx.transparent_bundle.is_some(), txid_parts.transparent_digests.is_some());
```
https://www.cs.utexas.edu/users/EWD/transcriptions/EWD10xx/EWD1070.html :-)